Creation of AIs and silicon law changes are now reported in deadchat

BadSS13Player opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR adds deadchat notifications for law changes of AIs and unsynced borgs ~(listing the full new lawset)~, uses of AI/Cyborg Upload and the creation of new AIs.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The AI is one of the most important players in the round, so it only makes sense that players watching in deadchat are kept up to date on its laws, especially hidden laws since they are never stated.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Silicon law changes are now reported in deadchat. 
add: The creation of new AIs is now reported in deadchat.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BadSS13Player committed:
Creation of AIs and silicon law changes are now reported in deadchat

carshalash commented:
Shift click the ai and borgs, nerd.

BadSS13Player committed:
Update AI_modules.dm

BadSS13Player committed:
Update ai_core.dm

cacogen commented:
>listing the full new lawset

is that a good idea given deadchat is already pretty overwhelmed with information and people's tendency to spam? would it be better to add a link to the message that prints the laws to chat if clicked?

BadSS13Player commented:
> would it be better to add a link to the message that prints the laws to chat if clicked?

I could make it a link to print the current laws, would be a little trickier to have it print the laws which were there at that moment (embed all the laws in the link?).

cacogen commented:
I dunno whatever is easiest for you I just figured this could be spammy

On Fri, 13 Dec 2019 at 9:09 AM, BadSS13Player <notifications@github.com>
wrote:

> would it be better to add a link to the message that prints the laws to
> chat if clicked?
>
> I could make it a link to print the current laws, would be a little
> trickier to have it print the laws which were there at that moment (embed
> all the laws in the link?).
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/48216?email_source=notifications&email_token=AF7NO6T4MEMSXTHNXQ7EZHLQYKK7HA5CNFSM4JZVXHM2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEGX37BI#issuecomment-565165957>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF7NO6VNK6TNLXGUFNEFKVTQYKK7HANCNFSM4JZVXHMQ>
> .
>
-- 
Sent from my phone

optimumtact commented:
add a pref for it like the ghost ears/ghost vision

maybe ghost antag tracking or something. Not sure what a descriptive name for such a pref would be

BadSS13Player committed:
Make the deadchat law changes a link and add a chat preference

BadSS13Player commented:
I made it so it no longer displays all the laws but instead displays a link that sends all the laws to chat.
I also made a chat preference for this, and made it not show the law changes or upload console use if off. Should I also include AI creation or should that be in ARRIVALRATTLE?

BadSS13Player committed:
Update laws.dm

BadSS13Player committed:
whoops

BadSS13Player committed:
whoops 2

BadSS13Player committed:
Fix the thing not working for AIs

This actually does change some small stuff for the AI. Namely, the AI can now use the state laws dialog while incapacitated, but that doesn't really affect gameplay and it's like that for cyborgs already.

BadSS13Player committed:
webedit bruh moment

BadSS13Player commented:
Here's a screenshot:
![ai law change screenshot](https://user-images.githubusercontent.com/58376109/71093445-1b518500-21a1-11ea-8fce-7de34ca43a57.png)
